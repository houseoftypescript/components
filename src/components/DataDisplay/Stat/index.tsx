import styled from 'styled-components';

export const Stat = styled.div`
  border: 1px solid #d1d5db;
  border-radius: 0.25rem;
  overflow: hidden;
  padding: 1rem;
  display: flex;
  flex-direction: column;
  gap: 0.25rem;
`;

export const StatLabel = styled.div`
  font-weight: 500;
  font-size: 0.875rem;
`;

export const StatNumber = styled.div`
  vertical-align: baseline;
  font-weight: 600;
  font-size: 1.5rem;
  font-variant-numeric: proportional-nums;
`;

export const StatHelpText = styled.div`
  opacity: 0.8;
  font-size: 0.875rem;
`;

export default Object.assign(Stat, {
  Label: StatLabel,
  Number: StatNumber,
  HelpText: StatHelpText,
});
