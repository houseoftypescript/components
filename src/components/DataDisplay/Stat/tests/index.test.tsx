import { render } from '@testing-library/react';
import React from 'react';
import Stat from '../index';

describe('Stat', () => {
  it('to match snapshot', () => {
    const { container } = render(
      <Stat>
        <Stat.Label>Collected Fees</Stat.Label>
        <Stat.Number>Â£0.00</Stat.Number>
        <Stat.HelpText>Feb 12 - Feb 28</Stat.HelpText>
      </Stat>
    );
    expect(container).toMatchSnapshot();
  });
});
