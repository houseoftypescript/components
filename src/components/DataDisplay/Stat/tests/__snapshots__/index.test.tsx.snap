// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stat to match snapshot 1`] = `
<div>
  <div
    class="sc-beySbM hcPYSl"
  >
    <div
      class="sc-guDLRT bCVCfQ"
    >
      Collected Fees
    </div>
    <div
      class="sc-dmyDGi jniSYs"
    >
      Â£0.00
    </div>
    <div
      class="sc-hLQRIN hfEdmM"
    >
      Feb 12 - Feb 28
    </div>
  </div>
</div>
`;
